neuralop/layers/fourier_continuation.py

```diff
@@ -85,9 +108,206 @@ def extend2d(self, x, one_sided):
         x = self.extend_top_bottom(x, one_sided)
 
         return x
+
+    def extend3d(self, x, one_sided):
+        x = self.extend_left_right(x, one_sided)
+        x = self.extend_top_bottom(x, one_sided)
+        x = self.extend_front_back(x, one_sided)
+        return x
     
     def forward(self, x, dim=2, one_sided=True):
         if dim == 1:
             return self.extend1d(x, one_sided)
         if dim == 2:
             return self.extend2d(x, one_sided)
+        if dim == 3:
+            return self.extend3d(x, one_sided)
+
+
+
+
+class FCGram(nn.Module):
+    """
+    FCGram class for Fourier Continuation using Gram matrices.
+    
+    This class implements the FC-Gram algorithm described in Section 3.1 of:
+    Amlani, F., & Bruno, O. P. (2016). An FC-based spectral solver for
+    elastodynamic problems in general three-dimensional domains. 
+    Journal of Computational Physics, 307, 333-354.
+    
+    The algorithm uses pre-computed Gram matrices (ArQr, AlQl) to perform
+    Fourier Continuation of discretized functions.
+    """
+    
+    def __init__(self, d, n_additional_pts, matrices_path='./FCGram_matrices'):
+        """
+        Initialize FCGram with specified parameters.
+        
+        Args:
```

Comment:
our convention has been Numpy docstrings - check the Contributor's section of `README.rst`